tools/appimage/build.sh

```diff
@@ -0,0 +1,207 @@
+	mv squashfs-root appimagetool
+}
+build-obtain-python() {
+	# Obtain the version of Python we need to build Randovania
+	# This extracst the image to ./squashfs-root
+	wget "$python_appimage"
+	chmod +x python*-manylinux1_x86_64.AppImage
+	./python*-manylinux1_x86_64.AppImage --appimage-extract
+}
+build-obtain-nuget() {
+	# Obtain the version of nuget we need to grab Mono app dependencies
+	wget "$nuget_url"
+	mkdir -p nuget
+}
+build-copy-randovania-into-image() {
+	# Create a directory in the working squashfs-root and sync the source tree
+	# into it.
+	mkdir -p squashfs-root/"$randovania_location"
+	rsync -a --exclude={"$workdir"/,"$outdir"/} ../ squashfs-root/"$randovania_location"
+}
+build-copy-overlay-into-image() {
+	# Copy the overlay (which contains AppImage and Linux-specific files) into the
+	# working squashfs-root
+	rsync -a ../tools/appimage/overlay/ squashfs-root/
+}
+build-static-mono() {
+	# Bundle up our Mono apps to avoid having to redistribute the Mono runtime
+	# This depends on a few things on the host:
+	#
+	#  - Libraries liblzo2 and libmono-native must both be present
+	#
+	#  - The .config files that add a dllmap for liblzo2 are required at build time
+	#    (this step is handled by copying in the overlay above)
+	#
+	#  - The host has to have mono-complete installed (package name is a debianism,
+	#    but the long and short of it is that you need the full runtime and dev
+	#    tooling
+	#
+	#  - The host has to have a libc version <= that of the oldest version we're
+	#    willing to support.
+	#
+	# If all goes well, you'll get a static binary that runs on anything, no Mono
+	# runtime required. No futzing with liblzo2 required. Just Werks.
+
+	# EchoesMenu.exe
+	pushd squashfs-root/"$randovania_location"/randovania/data/ClarisEchoesMenu
+	mkbundle $mono_mkbundle_args --config EchoesMenu.exe.config EchoesMenu.exe -o echoes-menu
+	popd
+
+	# Randomizer.exe
+	# This one is slightly more complex, as we have to add several dependencies
+	# via nuget.exe since they're either third-party or just straight up not
+	# included in the cross-compile runtimes for some reason
+	mono nuget.exe install NewtonSoft.json -OutputDirectory nuget
+	#mono nuget.exe install Novell.Directory.Ldap -OutputDirectory nuget
+	nugetdir="$PWD/nuget"
+	pushd squashfs-root/"$randovania_location"/randovania/data/ClarisPrimeRandomizer/
+	mkbundle $mono_mkbundle_args --config Randomizer.exe.config Randomizer.exe -o randomizer \
+		-L "$nugetdir"/Newtonsoft.Json.*/lib/net45 \
+		-L "$nugetdir"/Novell.Directory.Ldap.*/lib
+	popd
+}
+build-install-randovania() {
+	# Enter the Randovania directory inside squashfs-root and install its
+	# dependencies. We avoid activating the venv here because the Python
+	# interpreter in the AppImage we extracted already has a pseudo-venv going on
+	# with respect to the AppImage root.
+	pushd squashfs-root/"$randovania_location"
+	# Install dependencies
+	python -m pip install --upgrade -r requirements-setuptools.txt
```

Comment:
This entire function would be just using the executable from `create_release.py`.